recipes/gdal/post_3.5.0/conanfile.py

```diff
@@ -0,0 +1,889 @@
+    name = "gdal"
+    description = "GDAL is an open source X/MIT licensed translator library " \
+                  "for raster and vector geospatial data formats."
+    license = "MIT"
+    topics = ("osgeo", "geospatial", "raster", "vector")
+    homepage = "https://github.com/OSGeo/gdal"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    settings = "os", "arch", "compiler", "build_type"
+
+    generators = "cmake", "cmake_find_package"
+
+    # A list of gdal dependencies can be taken from cmake/helpers/CheckDependentLibraries.cmake
+    # within gdal sources with the command:
+    # grep -E '^[ \t]*gdal_check_package\(' cmake/helpers/CheckDependentLibraries.cmake \
+    #   | sed 's/[ \t]*gdal_check_package(\([a-zA-Z_0-9]\+\) "\(.*\)"\(.*\)/{ 'dep': \'\1\', 'descr': \'\2\' },/' \
+    #   | sort | uniq
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "tools": [True, False],
+        "with_armadillo": [True, False],
+        "with_arrow": [True, False],
+        "with_blosc": [True, False],
+        "with_cfitsio": [True, False],
+        "with_crypto": [True, False],
+        "with_cryptopp": [True, False],
+        "with_curl": [True, False],
+        "with_dds": [True, False],
+        "with_expat": [True, False],
+        "with_exr": [True, False],
+        "with_freexl": [True, False],
+        "with_geos": [True, False],
+        "with_gif": [True, False],
+        "with_gta": [True, False],
+        "with_hdf4": [True, False],
+        "with_hdf5": [True, False],
+        "with_heif": [True, False],
+        "with_kea": [True, False],
+        "with_libdeflate": [True, False],
+        "with_libiconv": [True, False],
+        "with_jpeg": [None, "libjpeg", "libjpeg-turbo"],
+        "with_libkml": [True, False],
+        "with_libtiff": [True, False],
+        "with_lz4": [True, False],
+        "with_mongocxx": [True, False],
+        "with_mysql": [None, "libmysqlclient", "mariadb-connector-c"],
+        "with_netcdf": [True, False],
+        "with_odbc": [True, False],
+        "with_openjpeg": [True, False],
+        "with_openssl": [True, False],
+        "with_pcre": [True, False],
+        "with_pcre2": [True, False],
+        "with_pg": [True, False],
+        "with_png": [True, False],
+        "with_podofo": [True, False],
+        "with_poppler": [True, False],
+        "with_proj": [True, False],
+        "with_qhull": [True, False],
+        "with_sqlite3": [True, False],
+        "with_webp": [True, False],
+        "with_xerces": [True, False],
+        "with_xml2": [True, False],
+        "with_zlib": [True, False],
+        "with_zstd": [True, False],
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "tools": False,
+        "with_armadillo": False,
+        "with_arrow": False,
+        "with_blosc": False,
+        "with_cfitsio": False,
+        "with_crypto": False, # deprecated, replaced by with_openssl
```

Comment:
Let's not add deprecated options on a new recipe

```suggestion
```

Reply:
well, this option exists in `recipes/gdal/pre_3.5.0/conanfile.py`. So for users upgrading to 3.5.1 and using --with_crypto, it would be nice to give a proper message.

I think --with_crypto is far less clear than --with_openssl, that's why I took the liberty to rename that option.
Would you prefer to carry around the bad --with_crypto name, that comes from outdated autoconf scripts in gdal < 3.5 ?
Or do you prefer that conan simply compains about an non-existing --with_crypto option?
Or are you OK with my solution: give a proper error message if --with_crypto is used?

Reply:
Holistically, I would like to follow upstreams build scripts as much as possible.

https://github.com/OSGeo/gdal/blob/e43b0c4f5ec61a1d66fb1f667a50b29380ce48a3/port/CMakeLists.txt#L256

Your names is correct 

> it would be nice to give a proper message.

Please put that in the comment. I think you should set it to `"ANY"` and only allow that value

This was it won't affect the package ID regardless

Reply:
Just remove it. Deprecated options can be removed after few versions. It's documented here: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#can-i-remove-an-option-from-a-recipe and it's a regular action removing deprecated stuff.

Reply:
Also, we have been removing deprecated options for many months, and so far, we never had a situation where someone opened an issue to complain about it.

Reply:
well, the option is not deprecated on master.
My suggestion is to deprecate it. You're suggesting to remove the option without any deprecation.
In the link you provide, I read:
>  It's preferable to leave all options (ie. not removing them) because it may break other packages which require those deleted options. Prefer the deprecation path with a mapping from old options to new ones

which is what I was trying to do here.

But I am not fighting anymore to make conan user friendly, I removed the option and just left a comment.

Reply:
I am not sure to follow you. Are you suggesting something like
```
options = { "with_crypto": ["deprecated: replaced by with_openssl"] }
```
or:
```
options = { "with_crypto": [ True, "deprecated"] }
def validate(self):
    if self.options.get_safe("with_crypto") != "deprecated":
        # complain with a fatal error
```

I do not understand very well how conan package ID works.

Reply:
There's a very good example in the docs (that were linked above)

https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#can-i-remove-an-option-from-a-recipe

Reply:
```py
options = {"with_crypto": [True, False, "deprecated"]}
default_options = {"with_crypto": "deprecated"}

def configure(self):
    if self.options.with_crypto != "deprecated":
        self.output.warn("with_crypto option is deprecated, use ???? instead.")

def package_id(self):
    del self.info.options.with_crypto
```